apps/ejabberd/src/mongoose_lib.erl

```diff
@@ -9,3 +11,26 @@ bin_to_int(<<H, T/binary>>, X) when $0 =< H, H =< $9 ->
     bin_to_int(T, (X*10)+(H-$0));
 bin_to_int(Bin, X) ->
     {X, Bin}.
+
+
+%% @doc Database backends for various modules return ok, {atomic, ok}
+%% or {atomic, []} on success, and usually {error, ...} on failure.
+%% All we need is to log an error if such occurred, and proceed normally.
+-spec log_if_backend_error(V :: any(), % value return by called backend fun
+                           Module :: atom(), % caller
+                           Line :: integer(),
+                           Args :: any() ) -> ok.
+log_if_backend_error(V, Module, Line, Args) ->
+    case V of
+        ok -> ok;
+        {atomic, _} -> ok;
+        {error, E} ->
+            make_msg("Error calling backend", E, Module, Line, Args);
+        E ->
+            make_msg("Unexpected return from backend", E, Module, Line, Args)
+    end,
+    ok.
+
+make_msg(Msg, Error, Module, Line, Args) ->
+    ?ERROR_MSG("~p: ~p~nBackend called in ~p:~p~nwith arguments~n~p",
```

Comment:
I think it would be nice to have it in logfmt-compliant format. Or at least with no `~n`s.